### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by atticus-sullivan** [L3MON4D3/LuaSnip#594] [General]
Hi,
after working on one feature, I think I'd like to contribute more to this project (if you don't object ^^). Any guidelines on how to choose features/bugs/...? (do you use the assign issue functionality to indicate that someone is working on this issue already? EDIT: Or maybe this is creating too much pressure for just a hobby project)

Oh and two things for starters would be introducing a `autosnippet` subsection to the `snippet` section in the docs and maybe a `CONTRIBUTING.md` file for people like me (on the other hand, I hope this discussion servers a similar purpose)

**Comment by L3MON4D3** [comment]
>  I think I'd like to contribute more to this project

Ayyy nice, that's great to hear :D

> Any guidelines on how to choose features/bugs/...?

Hmm, I can't, with a clean conscience, recommend working on the internals, they are pretty messy (so bugs like "a restoreNode inside a dynamicNode messes up something")
But apart from that, anything that sounds interesting. I'll do my best to help if you have any questions :)

One specific recommendation: maybe the trigger-types #547, if that sounds interesting to you.
(although I have some ideas there already, so be prepared for some discussion :P)

> (do you use the assign issue functionality to indicate that someone is working on this issue already? EDIT: Or maybe this is creating too much pressure for just a hobby project)

No, not yet, but it's a good idea (definitely better than having to ask if somebody is working on this already/going through the issue only to see that somebody's already working at it)

> Oh and two things for starters would be introducing a autosnippet subsection to the snippet section in the docs and maybe a CONTRIBUTING.md file for people like me (on the other hand, I hope this discussion servers a similar purpose)

Ouh, a CONTRIBUTING.md sounds great :+1:

**Reply by atticus-sullivan** [reply]
Alright, so I'll just look at the issues and check if there's anything (you can also just mention me if you see something interesting ;)
(concerning working on internals, I didn't had a look so far, if it's just ugly lua stuff, I've got no problem with that, if it's `nvim` specific, I think I'll look at some other things first xD)

I'll have a look regarding #547 but again I'll need some time (especially checking out the capabilities of the `vim` library to understand what you're talking about in that issue ^^).

Apart from the issues, the docs are another place worth investing time, I guess.

**Reply by L3MON4D3** [reply]
> you can also just mention me if you see something interesting ;)

Nice, will keep that in mind :D
Internals are not pretty just because it's grown a bit out of control, broken windows basically.. It could do with some refactoring, and I'll have to document how everything even works.

>Apart from the issues, the docs are another place worth investing time, I guess.

Oh, docs always👍
I'm already tackling that a bit in #551, should be quite a bit nicer once that's done :D

**Reply by atticus-sullivan** [reply]
Oh ok, maybe some day I'll have a look and maybe doing some refactoring (but as I read it, this will take time ^^).

#551 looks massive on first glance (take it as a positive thing :+1:).
